IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -617,10 +619,43 @@ - (void)sendTokenRequestImpl:(MSIDRequestCompletionBlock)completionBlock
                 }
             }
             
+            BOOL disableRemoveAccountArtifacts = [MSIDFlightManager.sharedInstance boolForKey:MSID_FLIGHT_DISABLE_REMOVE_ACCOUNT_ARTIFACTS];
+
+            // remove account artifacts only if we test flight feature is not disabled
+            if (!result && !disableRemoveAccountArtifacts && [self shouldRemoveAccountArtifacts:localError])
+            {
+                MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Account deleted, Removing any user account artifacts from device...");
+                [self removeAccountArtifacts:self.requestParameters];
+            }
+
             completionBlock(result, localError);
         }];
     }];
 }
+
+- (void)removeAccountArtifacts:(MSIDRequestParameters *)requestParameters
+{
+    NSError *removalError = nil;
+    BOOL removalResult = [self.tokenCache clearCacheForAccount:requestParameters.accountIdentifier
+                                                     authority:requestParameters.authority
+                                                      clientId:requestParameters.clientId
+                                                      familyId:nil
+                                                       context:requestParameters
+                                                         error:&removalError];
+    if (!removalResult)
+    {
+        MSID_LOG_WITH_CTX_PII(MSIDLogLevelWarning, requestParameters, @"Failed to clear cache with error %@", MSID_PII_LOG_MASKABLE(removalError));
+    }
+    
+    NSError *metadataRemovalError = nil;
```

Comment:
are we proceeding with metadata removal even when account removal didnt succeed ?

Reply:
True, in this case user is deleted from ESTS and all removal for any artifacts is good.